bin/gaas_maker_merge_outputs_from_datastore.pl

```diff
@@ -270,7 +270,7 @@ sub collect_recursive {
  		#deal with gff #
     	if($suffix eq ".gff"){
     		system "awk -F '	' 'NF==9 {print \$0 >> \"$out/$maker_mix_prefix.gff\"}' $full_path";
-				system "awk '{if(\$2 ~ /[a-zA-Z]+/) if(\$2==\"maker\") { print \$0 >> \"$out/$maker_annotation_prefix.gff\" } else { gsub(/:/, \"_\" ,\$2); print \$0 >> \"$out/\"\$2\".gff\" } }' $full_path";
+				system "awk '{if(\$2 ~ /[a-zA-Z]+/ && \$2==\"maker\") { print \$0 >> \"$out/$maker_annotation_prefix.gff\" } if (\$2 ~ /[a-zA-Z]+/ && \$2!=\"maker\") { gsub(/:/, \"_\" ,\$2); print \$0 >> \"$out/\"\$2\".gff\" } }' $full_path";
```

Comment:
Not that I'm involved with this project, but why are you calling `awk` from within Perl? It seems to me that Perl should be able to do the same thing, just like you've done in the previous sections of code... At the moment, you have a code injection vulnerability in the `awk` code due to injecting the Perl variables `$out` and `$make_annotation_prefix` as code into `awk`. The variable `$full_path` would also cause issues as it's unquoted in the `system()` call (would break if containing spaces, semi-colons, ampersands, filename globbing patterns etc.)

Reply:
I was using awk because felt it was much faster for that task. If it not the case I guess it is welcome to be rewritten in pure perl.

Reply:
I did a quick test and it becomes significantly slow. So I think keeping the awk is a good strategy in here.